addons/website/static/src/js/tours/tour_utils.js

```diff
@@ -274,9 +274,18 @@ function registerEditionTour(name, options, steps) {
     let tourSteps = steps;
     let url = getClientActionUrl(options.url, !!options.edition);
     if (options.edition) {
-       tourSteps = [{
+        tourSteps = [{
+            content: "Wait for the web client to be ready",
+            trigger: 'body',
+            run: () => {}, // It's a check
+        }, {
+            content: "Wait for the iframe preview to be loaded (but not ready)",
+            trigger: 'iframe[is-ready="false"] *',
+            run: () => {}, // It's a check
+        }, {
+            content: "Wait for the edit mode to be started",
             trigger: '.o_website_preview.editor_enable.editor_has_snippets',
-            timeout: 30000,
+            timeout: 15000,
```

Comment:
Do it ! :)

Note TBH: I am not sure how splitting that step in 3 would reduce the race conditon, but if you are sure about it, why not :open_mouth: Also, about the second step is not there a way it is ready very quickly and the second step could thus fail ?

Reply:
That's what I explain in commit message, when the runbot and/or connection is slow, the time between "starting the tour" and "wait for edit mode to be started" (the frist check we do) is too long, we then have to insert a bigger timeout step.
Now, with this commit, we wait for intermediate steps, each time resetting the timeout counter (since we found a step, we check for the next one with a fresh timer).

So in a nutshell, before:
```
(t0) The main document's tour manager is ready (timeout starts for next step)
(+13) The editor started (timeout reset for next step)
```

Now: 
```
(t0) The main document's tour manager is ready (timeout starts for next step)
(+1) The web client is ready (timeout reset for next step)
(+9) The iframe's tour manager is ready (timeout reset for next step)
(+12) The iframe is ready (Public root ready) (timeout reset for next step)
(+13) The editor started (timeout reset for next step)
```

This PR is just dividing the "auto added step" in multiple steps to avoid reaching the timeout.
There is no gain whatsoever over using a bigger timeout to avoid the error.

So I guess this PR is a bit overkill, especially since I had to introduce a data-attribute on the DOM to know when the tour manager is ready in the iframe (maybe there is a better way).

Reply:
> Now, with this commit, we wait for intermediate steps, each time resetting the timeout counter (since we found a step, we check for the next one with a fresh timer).

Yep I understood all that.

> There is no gain whatsoever over using a bigger timeout to avoid the error.

That was my point. IMO splitting in multiple steps actually might increase the race condition and is definitely "slower". So if there is no gain, then I would not do it :grimacing: